### PR TITLE
Revert/Fix: epoch indexing from 1, to be from 0

### LINKED ISSUE #1946: ModelCheckpoint epoch and progress bar epoch mismatch
## 🐛 Bug

Progress bar epoch starts at 1, but in ModelCheckpoint it starts at 0.

### To Reproduce

https://github.com/PyTorchLightning/pytorch-lightning/blob/8ca8336ce52ee7379f4d399520636143eb31018b/pytorch_lightning/callbacks/progress.py#L320

https://github.com/PyTorchLightning/pytorch-lightning/blob/8ca8336ce52ee7379f4d399520636143eb31018b/pytorch_lightning/callbacks/model_checkpoint.py#L212

### Expected behavior

Shouldn't they be consistent?

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

This reverts commit f94b919b

Reaction to #1946
Fixes #2206

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/gradient_accumulation_scheduler.py [hunk 1]
```diff
@@ -42,12 +38,10 @@ def __init__(self, scheduling: dict):
 
         for key in scheduling:
             if not isinstance(key, int) or not isinstance(scheduling[key], int):
-                raise TypeError("All epochs and accumulation factor must be integers")
+                raise TypeError("All epoches and accumulation factor must be integers")
```
Comment (Borda) [inline pytorch_lightning/callbacks/gradient_accumulation_scheduler.py:41]:
```suggestion
                raise TypeError("All epochs and accumulation factor must be integers")
```

File: pytorch_lightning/callbacks/progress.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def total_val_batches(self) -> int:
         if trainer.fast_dev_run and trainer.val_dataloaders is not None:
             total_val_batches = len(trainer.val_dataloaders)
         elif not self.trainer.disable_validation:
-            is_val_epoch = trainer.current_epoch % trainer.check_val_every_n_epoch == 0
+            is_val_epoch = (trainer.current_epoch + 1) % trainer.check_val_every_n_epoch == 0
```
Comment (Borda) [inline pytorch_lightning/callbacks/progress.py:99]:
```suggestion
            is_val_epoch = trainer.current_epoch % trainer.check_val_every_n_epoch == 0
```

File: pytorch_lightning/trainer/training_loop.py [hunk 3]
```diff
@@ -476,7 +476,7 @@ def run_training_epoch(self):
             # RUN VAL STEP
             # ---------------
             is_val_check_batch = (batch_idx + 1) % self.val_check_batch == 0
-            can_check_epoch = self.current_epoch % self.check_val_every_n_epoch == 0
+            can_check_epoch = (self.current_epoch + 1) % self.check_val_every_n_epoch == 0
```
Comment (Borda) [inline pytorch_lightning/trainer/training_loop.py:479]:
```suggestion
            can_check_epoch = self.current_epoch % self.check_val_every_n_epoch == 0
```

File: tests/models/test_restore.py [hunk 4]
```diff
@@ -172,7 +172,7 @@ def test_dp_resume(tmpdir):
     result = trainer.fit(model)
 
     # track epoch before saving. Increment since we finished the current epoch, don't want to rerun
-    real_global_epoch = trainer.current_epoch
+    real_global_epoch = trainer.current_epoch + 1
```
Comment (Borda) [inline tests/models/test_restore.py:175]:
```suggestion
    real_global_epoch = trainer.current_epoch
```

**Comment by williamFalcon** [comment]
@Borda we need to index epoch from 0

**Comment by Borda** [comment]
> @Borda we need to index epoch from 0

yes, it is now... with this PR

**Pull request merged**